### PR TITLE
Map Azure public facing with public ip

### PR DESCRIPTION
**Pull request by ganeshjagdale9545**
**I have tested this logic with 2 vms and virtual network**

### CODE REVIEW

File: cartography/data/jobs/analysis/azure_network_public_facing_asset_exposure.json [hunk 1]
```diff
@@ -3,12 +3,11 @@
         {
             "query": "MATCH ((n) <-[:RESOURCE]-(:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})<-[:RESOURCE]-(:AzureTenant{id: {AZURE_TENANT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}})) where EXISTS(n.exposed_internet) AND labels(n) IN [['AzureNetwork']] WITH n LIMIT {LIMIT_SIZE} REMOVE n.exposed_internet return COUNT(*) as TotalCompleted",
             "iterative": true,
-            "iterationsize": 1000
+            "iterationsize": 100
```
Comment (mpurusottamc) [inline cartography/data/jobs/analysis/azure_network_public_facing_asset_exposure.json:6]: any specific reason we have reduced this from 1000 to 100?

File: cartography/data/jobs/analysis/azure_vm_public_facing_asset_exposure.json [hunk 2]
```diff
@@ -3,12 +3,15 @@
         {
             "query": "MATCH ((n) <-[:RESOURCE]-(:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})<-[:RESOURCE]-(:AzureTenant{id: {AZURE_TENANT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}})) where EXISTS(n.exposed_internet) AND labels(n) IN [['AzureVirtualMachine']] WITH n LIMIT {LIMIT_SIZE} REMOVE n.exposed_internet return COUNT(*) as TotalCompleted",
             "iterative": true,
-            "iterationsize": 1000
+            "iterationsize": 100
```
Comment (mpurusottamc) [inline cartography/data/jobs/analysis/azure_vm_public_facing_asset_exposure.json:6]: same as above.

**Review by mpurusottamc — APPROVED** [review]
looks good.

File: cartography/data/jobs/analysis/azure_vm_public_facing_asset_exposure.json (hunk 2 above)
Comment (mpurusottamc) [reply cartography/data/jobs/analysis/azure_vm_public_facing_asset_exposure.json:6]: never mind. ignore this.

File: cartography/data/jobs/analysis/azure_vm_public_facing_asset_exposure.json [hunk 3]
```diff
@@ -3,12 +3,15 @@
...
-            "iterative": true,
-            "iterationsize": 1000
+            "query": "MATCH ((rule:AzureNetworkSecurityRule)<-[:CONTAIN]-(group:AzureNetworkSecurityGroup)<-[:MEMBER_NETWORK_SECURITY_GROUP]-(interface:AzureNetworkInterface)<-[:MEMBER_NETWORK_INTERFACE]-(vm:AzureVirtualMachine)<-[:RESOURCE]-(:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})<-[:RESOURCE]-(:AzureTenant{id: {AZURE_TENANT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}))\nWHERE rule.access='Allow' And rule.source_port_range='*' OR rule.protocol='*' OR rule.source_address_prefix='0.0.0.0/0' \nSET vm.exposed_internet = true;",
+            "iterative": false
+        },
+        {
+            "query": "MATCH ((ip:AzurePublicIPAddress)<-[:MEMBER_PUBLIC_IP_ADDRESS]-(interface:AzureNetworkInterface)<-[:MEMBER_NETWORK_INTERFACE]-(vm:AzureVirtualMachine)<-[:RESOURCE]-(:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})<-[:RESOURCE]-(:AzureTenant{id: {AZURE_TENANT_ID}})<-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}}))\nSET vm.exposed_internet = true;",
+            "iterative": false
```
[the remark this answers is not in the document]
Comment (ganeshjagdale9545) [reply cartography/data/jobs/analysis/azure_vm_public_facing_asset_exposure.json:14]: if set iterative true then analysis job take lots of time and also for AWS they don’t have set  iterative true

**Pull request merged**